src/main/java/reactor/ipc/netty/http/client/MonoHttpClientResponse.java

```diff
@@ -183,7 +184,8 @@ public boolean test(Throwable throwable) {
 				redirect(re.location);
 				return true;
 			}
-			if (AbortedException.isConnectionReset(throwable)) {
+			if (!retried && AbortedException.isConnectionReset(throwable)) {
```

Comment:
micro optimization: we can invert the condition order to check the volatile only when AbortedException.

Reply:
done